### DISCUSSION TITLE
OpenAip Map Integration in C# (Visual Studio)

### DISCUSSION THREAD

**Discussion by SimoLP** [openAIP/openaip-public#320] [Q&A] [answered]
Hi, 
I'm an aspiring developer.
I want to use the OpenAIP map for a VFR flight planning app in C# (Visual Studio).
Having never done this type of work, I was wondering if someone could explain to me how I can integrate this map into Visual Studio C# so that the user can use it and select points to make routes (so you will have to see a line representing the route from one point to another).

Regards

SimoLP

**Comment by reskume** [comment, marked as answer]
Hi Simon,

this is a rather big topic to answer since it involves knowing the used technology stack and libraries involved. In general, you have to find a library that can either display an TMS PNG or MVT binary tile in your application. Then, depending on the library's API, you can draw different things on the map. Hope this gives at least a general idea of how I would do it :)

Cheers,

Stephan

**Reply by SimoLP** [reply]
Thank you.

If you know one, can you recommend me a suitable library for OpenAIP Integration?

**Reply by reskume** [reply]
Since I never worked with c#, I cannot provide much support here. We use JS exclusively and there are plenty of libraries you can use like [Mapbox GL JS](https://docs.mapbox.com/mapbox-gl-js/api/) or [OpenLayers](https://openlayers.org/). We use Mapbox GL JS for showing every geocontent on openaip.
